libsolidity/inlineasm/AsmData.h

```diff
@@ -49,9 +49,11 @@ struct Assignment { SourceLocation location; Identifier variableName; };
 struct FunctionalAssignment;
 struct VariableDeclaration;
 struct FunctionDefinition;
+struct Switch;
 struct FunctionCall;
 struct Block;
-using Statement = boost::variant<Instruction, Literal, Label, Assignment, Identifier, FunctionalAssignment, FunctionCall, FunctionalInstruction, VariableDeclaration, FunctionDefinition, Block>;
+using Statement = boost::variant<Instruction, Literal, Label, Assignment, Identifier, FunctionalAssignment, FunctionCall, FunctionalInstruction, VariableDeclaration, FunctionDefinition, Switch, Block>;
+using Expression = boost::variant<FunctionCall, Identifier, Literal>;
```

Comment:
I we introduce another variant, our AST walker will get more complicated (because we need two types of them, and there is no unified way to walk the whole AST). We can force this restriction inside the parser or some other component.